ipmof/crystal.py

```diff
@@ -4,27 +4,46 @@
 import math
 import os
 from ipmof.forcefield import get_ff_parameters
+import ipmof.io as io
 
 
 class MOF:
     """
     MOF class that holds coordinate, atom name, and unit cell information
     """
-    def __init__(self, mof_file, file_format='mol2'):
+    def __init__(self, file_path, file_format='cif', reader='ase'):
         """
         Initialize MOF name, unit cell volume and parameters, atom names and coordinates, and
-        unique atom names and coordinates by reading values from mol2 file.
+        unique atom names and coordinates.
+        reader -> ('babel' / 'ase')
+        file_format -> ('cif' / 'dict' / 'mol2' / ...)
         """
-        if file_format == 'mol2':
-            self.mol2_path = mof_file
-            self.name = os.path.split(self.mol2_path)[-1].split('.')[0]
-            self.read_mol2()
+        if file_format == 'dict':
+            molecule = file_path
+            self.atom_coors = molecule['atom_coors']
+            self.atom_names = molecule['atom_names']
+            self.name = molecule['name']
+        else:
+            file_format = os.path.splitext(file_path)[-1][1:]
+            self.path = file_path
+            self.name = os.path.split(self.path)[-1].split('.')[0]
+            if file_format == 'mol2':
+                molecule = io.read_mol2(file_path)
+                self.reader = 'mol2'
+            elif reader == 'ase':
+                ase = io.ase(input_format=file_format)
+                molecule = ase.read(file_path)
+                self.reader = 'ase'
+            elif reader == 'babel':
+                babel = io.babel(input_format=file_format)
+                molecule = babel.read(file_path)
+                self.reader = 'babel'
```

Comment:
something like this could work: 

reader = io.babel 
reader = io.ase

r = reader(input_format=file_format)
molecule = f.read(file_path)